### PR TITLE
Check foreground from browser window visibility instead of Qt.application.active

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

File: src/browser.qml [hunk 1]
```diff
@@ -21,6 +21,7 @@ ApplicationWindow {
     id: window
 
     readonly property bool largeScreen: Screen.sizeCategory > Screen.Medium
+    readonly property int browserVisibility: _mainWindow && _mainWindow.visibility || QuickWindow.Window.Hidden
```
Comment (jpetrell) [inline src/browser.qml]: _mainWindow.visibility || QuickWindow.Window.Hidden looks weird?

File: src/pages/components/TabView.qml [hunk 2]
```diff
@@ -9,7 +9,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import QtQuick 2.1
+import QtQuick 2.2
+import QtQuick.Window 2.2 as QuickWindow
```
Comment (pvuorela) [inline src/pages/components/TabView.qml]: Not used here?

File: src/browser.qml (hunk 1 above)
Comment (pvuorela) [reply src/browser.qml]: "_mainWindow ? _mainWindow.visibility : QuickWindow.Window.Hidden" perhaps?

Comment (rainemak) [reply src/browser.qml]: I tend to use that notation. Will check to ternary operator.

File: src/pages/BrowserPage.qml [hunk 3]
```diff
@@ -40,7 +40,7 @@ Page {
     }
 
     function bringToForeground(window) {
-        if (!Qt.application.active && window) {
+        if ((webView.visibility == QtWindow.Window.Minimized) && window) {
```
Comment (pvuorela) [inline src/pages/BrowserPage.qml]: How would webView.visible work generally? Shorter and perhaps takes care of Window.Hidden?

Comment (rainemak) [reply src/pages/BrowserPage.qml]: It's still visible when on cover but updated a bit other way round.

File: src/pages/BrowserPage.qml [hunk 4]
```diff
@@ -40,7 +40,7 @@ Page {
     }
 
     function bringToForeground(window) {
-        if (!Qt.application.active && window) {
+        if ((webView.visibility < QuickWindow.Window.Maximized) && window) {
```
Comment (rainemak) [inline src/pages/BrowserPage.qml]: I don't think that QWindow::AutomaticVisibility could cause issue here.

Comment (pvuorela) [reply src/pages/BrowserPage.qml]: As small note conceptually raise() should be valid operation with everything else than fullscreen. Or maybe even that if considering app being fullscreen and window app on top of it. Method here doesn't look like getting called too much so could even remove window state checks, though doesn't matter much.

File: src/pages/components/Overlay.qml [hunk 5]
```diff
@@ -523,6 +524,7 @@ Background {
                 model: webView.tabModel
                 portrait: tabPage.isPortrait
                 privateMode: webView.privateMode
+                pullDownVisible: webView.visibility >= QuickWindow.Window.Maximized
```
Comment (pvuorela) [inline src/pages/components/Overlay.qml]: A little curious why this is done. Tab page per se not included in cover? Introduced 2014 with commit "Hide pulley menu, close tab button, title, and the opacity ramp in cover"

Comment (rainemak) [reply src/pages/components/Overlay.qml]: Ah, so true :+1: Will cleanup.

**Pull request merged**